File: backend/src/main/java/com/ecommerce/platform/service/RoleService.java
package com.ecommerce.platform.service;

import com.ecommerce.platform.entity.Role;
import java.util.Optional;
import java.util.List;
import java.util.Set;

public interface RoleService {
    Optional<Role> findByName(String name);

    Role getRoleByName(String name);

    Role getRoleById(Long id);

    List<Role> getAllRoles();

    Role createRole(Role role);

    Role updateRole(Long id, String description);

    void deleteRole(Long id);
}

File: frontend/src/services/roleService.js
import axiosClient from "./api";

const roleService = {
    getAllRoles: async () => {
        const response = await axiosClient.get("/roles");
        return response.data;
    },

    getRoleById: async (id) => {
        const response = await axiosClient.get(`/roles/${id}`);
        return response.data;
    },

    createRole: async (roleData) => {
        const response = await axiosClient.post("/roles", roleData);
        return response.data;
    },

    updateRole: async (id, roleData) => {
        const response = await axiosClient.put(`/roles/${id}`, roleData);
        return response.data;
    },

    deleteRole: async (id) => {
        const response = await axiosClient.delete(`/roles/${id}`);
        return response.data;
    },


};

export default roleService;
